Don't sleep and only wake when needed

mawdegroot opened the pull request.

mawdegroot committed:
Don't sleep and only wake when needed

madninja commented on src/ecc.rs:
```diff
@@ -265,7 +268,7 @@ impl Ecc {
             command.bytes_into(&mut buf);
 
-            self.transport.send_wake(wake_delay)?;
+            self.transport.send_wake(wake_delay, Duration::from_millis(self.config.min_wake_interval.into()))?;
```
* Move Duration::from to out of loop (below wake_delay)
* Use from_micros for consistency with the other timing values

madninja commented on src/transport.rs:
```diff
@@ -72,20 +73,30 @@ impl TransportProtocol {
 
 impl I2cTransport {
     pub fn new(path: &str, address: u16) -> Result<Self> {
+        let last_wake = Instant::now().checked_sub(Duration::from_secs(10)).unwrap();
```
Do _not_ use unwraps.. Use an Option<Instant> if you're trying to represent something in the past?

madninja commented on src/transport.rs:
```diff
@@ -72,20 +73,30 @@ impl TransportProtocol {
+            self.send_idle();
+            let _ = self.send_buf(0, &[0x00]);
+            thread::sleep(wake_delay);
+            self.last_wake = Instant::now();
```
Heh so the question is whether you set last_wake _after_  or _before the thread::sleep

madninja commented on src/transport.rs:
```diff
@@ -72,20 +73,30 @@ impl TransportProtocol {
+
     fn send_sleep(&mut self) {
-        let _ = self.send_buf(self.address, &[1]);
+        // disable sleeping and do a wake as needed instead
```
Instead of this we should probably set sleep to off for the calls that expect it? Or just remove the sleep true/false completely from the send_commands implementation

mawdegroot committed:
Going a different route with avoiding sleep

Use IDLE as much as possible, use SLEEP to reset the chip to a sane
state in the case of errors

mawdegroot committed:
remove remaining remnants of original idea

mawdegroot committed:
make cargo fmt happy
